tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -434,8 +438,10 @@ struct CursorInfoData {
   /// For calls this lists the USRs of the receiver types (multiple only in the
   /// case that the base is a protocol composition).
   ArrayRef<StringRef> ReceiverUSRs;
+  /// Secondary results, eg. the init when on the type in a constructor call or
+  /// getters/setters for a property reference.
+  llvm::SmallVector<CursorSymbolInfo, 1> Secondaries;
```

Comment:
FWIW, I really don't like this name - if anyone has something better please let me know!

Reply:
No great ideas on "secondary" ("additional" is about the same information content),  but I think we should at least make it `SecondarySymbols`

Reply:
Also why SmallVector<, 1> instead of ArrayRef?  This whole struct is designed to not own its storage and I find it weird that we now have such a mix of owned and unowned strings and arrays.

Reply:
All SmallString fields are now StringRef and there's now just `Symbols` inside `CursorInfoData`. `CursorInfoSymbol` owns its vectors though - I can change that by passing in a couple vector buffers (ie. one for StringRef and one for ParentInfo) but that really doesn't seem worth it to me.

I still find the StringRef's questionable, the only strings that are owned elsewhere are: `ModuleName`, `ModuleInterfaceName`,`Filename`, and `KindName` inside `ParentInfo`. `ModuleName` is sometimes constructed anyway (for a Clang decl) and `KindName` could probably (?) be a UIdent anyway. So that leaves `ModuleInterfaceName` and `Filename`, which could just be copied. `CursorSymbolInfo` could then have all `SmallString` instead of `StringRef`. 🤷